bundles/org.openhab.binding.luxtronikheatpump/src/main/java/org/openhab/binding/luxtronikheatpump/internal/enums/HeatpumpChannel.java

```diff
@@ -0,0 +1,784 @@
+            HeatpumpVisibility.THERMDESINFEKT),
+
+    // Comfort Kühlung Betriebsart
+    CHANNEL_EINST_BWSTYP_AKT(100, "Einst_BWStyp_akt", NumberItem.class, null, true, HeatpumpVisibility.KUHLUNG),
+
+    // Comfort Kühlung AT-Freigabe
+    CHANNEL_EINST_KUCFTL_AKT(110, "Einst_KuCft1_akt", NumberItem.class, SIUnits.CELSIUS, true,
+            HeatpumpVisibility.KUHLUNG),
+
+    // Comfort Kühlung AT-Freigabe Sollwert
+    CHANNEL_SOLLWERT_KUCFTL_AKT(132, "Sollwert_KuCft1_akt", NumberItem.class, SIUnits.CELSIUS, true,
+            HeatpumpVisibility.KUHLUNG),
+
+    // AT-Überschreitung
+    CHANNEL_EINST_KUHL_ZEIT_EIN_AKT(850, "Einst_Kuhl_Zeit_Ein_akt", NumberItem.class, null, true,
+            HeatpumpVisibility.SYSEIN_KUHL_ZEIT_EIN),
+
+    // AT-Unterschreitung
+    CHANNEL_EINST_KUHL_ZEIT_AUS_AKT(851, "Einst_Kuhl_Zeit_Aus_akt", NumberItem.class, null, true,
+            HeatpumpVisibility.SYSEIN_KUHL_ZEIT_AUS);
+
+    private @Nullable Integer channelId;
+    private String command;
+    private Class<? extends Item> itemClass;
+    private @Nullable Unit<?> unit;
+    private Boolean isParameter;
+    private @Nullable HeatpumpVisibility requiredVisibility;
+
+    private HeatpumpChannel(@Nullable Integer channelId, String command, Class<? extends Item> itemClass,
+            @Nullable Unit<?> unit, Boolean isParameter, @Nullable HeatpumpVisibility requiredVisibility) {
+        this.channelId = channelId;
+        this.command = command;
+        this.itemClass = itemClass;
+        this.unit = unit;
+        this.isParameter = isParameter;
+        this.requiredVisibility = requiredVisibility;
+    }
+
+    public @Nullable Integer getChannelId() {
+        return channelId;
+    }
+
+    public String getCommand() {
+        return command;
+    }
+
+    public Class<? extends Item> getItemClass() {
+        return itemClass;
+    }
+
+    public @Nullable Unit<?> getUnit() {
+        return unit;
+    }
+
+    public Boolean isWritable() {
+        return isParameter == Boolean.TRUE;
+    }
+
+    protected @Nullable HeatpumpVisibility getVisibility() {
+        return requiredVisibility;
+    }
+
+    public Boolean isVisible(Integer[] visibilityValues) {
+        HeatpumpVisibility visiblity = getVisibility();
+
+        if (visiblity == null) {
+            return Boolean.TRUE;
+        }
+
+        int code = visiblity.getCode();
+
+        if (visibilityValues.length < code || visibilityValues[code].equals(1)) {
+            return Boolean.TRUE;
+        }
+
+        return Boolean.FALSE;
+    }
+
+    public static HeatpumpChannel fromString(String heatpumpCommand) {
+
+        for (HeatpumpChannel c : HeatpumpChannel.values()) {
+
+            if (c.getCommand().equals(heatpumpCommand)) {
+                return c;
+            }
+        }
+
+        throw new IllegalArgumentException("cannot find LuxtronikHeatpump channel for '" + heatpumpCommand + "'");
```

Comment:
Unchecked exception

Reply:
Unchecked does not mean it's missing in throws section of method signature. Instead it does mean the compiler does not force you to write a try block around it, but that would be better. Instead of reusing `InvalidArgumentExcpetion` use a different exception type or define you own. (also for the other places.)

Reply:
ok. will replace that with a custom exception